ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -378,20 +328,21 @@ func convertToContainerDef(context *project.Context, inputCfg *config.ServiceCon
 	// Only set shmSize if specified. Otherwise we expect this sharedMemorySize for the
 	// containerDefinition to be null; Docker will by default allocate 64M for shared memory if
 	// shmSize is null.
-	if shmSize != 0 {
-		outputContDef.LinuxParameters.SetSharedMemorySize(shmSize)
+	if inputCfg.ShmSize != 0 {
+		outputContDef.LinuxParameters.SetSharedMemorySize(inputCfg.ShmSize)
 	}
 
 	// Only set tmpfs if tmpfs mounts are specified.
-	if tmpfs != nil {
-		outputContDef.LinuxParameters.SetTmpfs(tmpfs)
+	if inputCfg.Tmpfs != nil { // will never be nil?
```

Comment:
Tmpfs is a pointer to a slice of ecs.Tmpfs in containerconfig.go - why are you thinking this may never be nil?

Reply:
I think I was recalling a conversation in which we talked about nil-checking either here vs in the new `parseVX` code and thinking that this would always be an empty slice if the incoming ServiceConfig was also empty/nil.

Reply:
oh! you're right, and I had it backwards - ContainerConfig.Tmpfs is a slice of pointers, so it might be empty but never nil.